File: go/p0001/solution.go
package p0001

func twoSum(nums []int, target int) []int {
	idx := map[int]int{}
	for i, n := range nums {
		if j, ok := idx[target-n]; ok {
			return []int{j, i}
		}
		idx[n] = i
	}
	return nil // ??
}

File: lua/p0001/solution.lua
return {
	two_sum = function(nums, target)
		local num_map = {}
		for i, num in ipairs(nums) do
			local j = target - num
			if num_map[target - num] then
				return {num_map[j], i}
			end
			num_map[num] = i
		end
		return nil -- ??
	end,
}